### PR TITLE
Debug highlighting

### PR DESCRIPTION
**Pull request by supreme2580**
- [ ] issue #
- [ ] follows contribution [guide](https://github.com/keep-starknet-strange/shinigami/blob/main/CONTRIBUTING.md)
- [ ] code change includes tests

Added text highlighting in debug mode so a user knows what op_code the debug data is for

### CODE REVIEW

**Review by b-j-roberts — COMMENTED** [review]
Like how it looks, but it seems to highlight any occurrence of an opcode.

For example, this script: "OP_1 OP_2 OP_ADD OP_3 OP_EQUAL OP_HASH160 OP_1 OP_2 OP_ADD OP_3"
In debug mode will highlight both OP_1 opcodes after clicking forward once

**Comment by supreme2580** [comment]
@b-j-roberts the tokenizer in monaco react do not supporting tokenizing a single word when it occur multiple times... that's why it is tokenizing both OP_1s

File: frontend/src/components/script-editor.tsx [hunk 1]
```diff
@@ -110,9 +116,50 @@ export default function ScriptEditor() {
...
+
+      if (step >= 0 && step < words.length) {
+        const wordToHighlight = words[step];
+        const matches = editorModel.findMatches(wordToHighlight, false, false, false, null, false);
+
+        if (matches.length > 0) {
+          let matchIndexToUse = 0;
+
+          if (matches.length > 1) {
+            const occurrencesSoFar = words.slice(0, step + 1).filter(word => word === wordToHighlight).length;
```
Comment (b-j-roberts) [inline frontend/src/components/script-editor.tsx:141]: This works for now, but there will be script layouts that we will want to support later that will be a lot harder to do like this. Could you see if you can create a `range` value programmatically without using findMatches?
something like `let range = {start: words[step], end: words[step].len}` ( It wont work exactly like this, but I think it shows what I mean )

**Review by b-j-roberts — CHANGES_REQUESTED** [review]

File: frontend/src/components/script-editor.tsx (hunk 1 above)
Comment (supreme2580) [reply frontend/src/components/script-editor.tsx:141]: I will look into this asap and get back to you

**Review by b-j-roberts — APPROVED** [review]
This is super cool! Thank you!

**Pull request merged**